### ISSUE TITLE
Can not get a good result

### ISSUE THREAD

**Issue by lucasliunju** [HeshamMeneisi/AnnaPacman#1]
I'm very sorry to trouble you! I'm running the code and it's very cool! But I can not get a good result after 30 hours. The agent can't learn some  skills and not intelligent! I'm looking forward to your replay.

**Comment by HeshamMeneisi** [comment]
Hello Lucas. Can you please run env_test.py and post the result here? Also, how many iterations were completed during these 30 hours?

Here are the times it took in 2 different environments as far as I can remember:

Using TensorFlow on i7 3630QM 2.4GHz it took a week and never finished level 1. But sensible behavior started after around 20 hours which I think was around 70000 iterations.

Using Theano on GTX 1070 it took less than a day and it would occasionally finish level 1 and even sometimes survive in level 2 for a minute or so.

The first intelligent behavior to appear was almost always collecting pellets in all directions on an intersection. The agent freezes in the middle of the intersection by pressing 3 different arrows quickly to collect all surrounding pellets.

I'm waiting for your reply with the **environment log** and the **number of iterations** to be able to help you.

**Comment by lucasliunju** [comment]
Hi Hesham ! Thanks for your replay.  I'm using Tensorflow on GTX 1080Ti. And the number of iterations almost 1,000,000. I find the agent can not escape when the ghost is close, which is not intelligent. At the same time, I find the loss always 0 when training.

**Comment by HeshamMeneisi** [comment]
I had already fixed this but apparently, it was lost. I did lose a commit before so I will check all the bugs I can remember.

There was indeed some bugs and misconfiguration. I have fixed everything I could see, thank you for flagging this. It should work now without any issues if you pull my last commit. Please tell me if the issue persists.

Also regarding ghosts, if you want it to effectively chase vulnerable ghosts but avoid normal ghosts you have to train it on RGB because the greyscale does not have enough information. With the current reward settings, it should avoid all ghosts.

**Comment by lucasliunju** [comment]
Thanks for your reply.  I'll try the new version and thanks for your help.I compared with the code's environment and MsPacman of gym. I find that the objects in gym is more bigger, such as the wall, beans. I think the size of object play a great role in CNN. If it's very small, the agent can not get enough information. I also think reward function is important.

Thanks for your help again!

**Comment by HeshamMeneisi** [comment]
No problem Lucas, tell me if there are any other issues and I'll work with you through them.

Regarding the input size, you can plot a frame using the commented debugging function in dqn.py to see what the agent sees (rotated and flipped to correct orientation). Additionally, all of the rewards are defined at the top of the file "game/pacman_container.py" feel free to temper with this configuration. 

I do agree with you. This game is much more complex than the gym version. The big board, shaded spirits and using only greyscale makes it a much more challenging task. However, I have successfully trained this model to pass the first level around 1 in 5 times.

If you achieve better results feel free to make a pull request with your configuration.

**Comment by lucasliunju** [comment]
Due to some objective factors, my code is slow and memory resources are not enough. So I would like to ask if you can share the model weight file.

**Comment by HeshamMeneisi** [comment]
Sorry for the late reply. I'm afraid that I don't have trained weights at hand right now I should have included the last working file in a commit. I hope you managed to achieve something.
